packages/context/src/app/AppProvider.tsx

```diff
@@ -62,8 +78,57 @@ export type AppProviderProps = {
 
 function AppProvider(props: AppProviderProps) {
     const { app, children } = props;
-
-    return <AppContext.Provider value={app}>{children}</AppContext.Provider>;
+    const apiServices = useApiServices();
+
+    const appWithFunctions: AppWithFunctions = {
+        ...app,
+        /**
+         * Looks up the default page slug for a given object or its nearest type ancestor.
+         *
+         * @param {string} objectId - The ID of the object to start the search from.
+         * @returns {Promise<string | undefined>} The default page slug, or `undefined` if no default page is found.
+         */
+        findDefaultPageSlugFor: useCallback(
+            async (objectId: string) => {
+                let defaultPageId: string | undefined;
+                const objectHierarchy = [objectId];
+                let currentObjectId: string | undefined = objectId;
+                while (currentObjectId !== undefined) {
+                    if (app.defaultPages?.[currentObjectId]) {
+                        defaultPageId = app.defaultPages[currentObjectId];
+                        break;
+                    }
+
+                    let effectiveObject: Obj | undefined;
+                    try {
+                        effectiveObject = await apiServices.get<Obj>(`data/objects/${currentObjectId}/effective`, {
+                            params: { filter: { fields: ['baseObject'] }, sanitizedVersion: true },
+                        });
+                    } catch (error) {
+                        console.error(error);
+                    }
+                    if (effectiveObject?.baseObject?.objectId) {
+                        objectHierarchy.push(effectiveObject?.baseObject?.objectId);
+                    }
+                    currentObjectId = effectiveObject?.baseObject?.objectId ?? undefined;
+                }
+
+                let defaultPage: Page | undefined;
+                if (defaultPageId && typeof defaultPageId === 'string') {
+                    const pageId = defaultPageId.includes('/')
+                        ? defaultPageId.split('/').slice(2).join('/')
```

Comment:
So I guess we're prepending the app slug to the page slug?  I suppose it doesn't matter too much now since we're moving to autogenerated ids, but I've seen that in other places and am curious why we're doing that?  Seems counter-intuitive.

Reply:
That's how the default pages are saved. I'm not sure why that was decided. Why is it counter-intuitive?

Reply:
It's not part of the page id/slug.  I noticed it with `navigateTo`, I think it expects you to include the app prefix in the navigation target, but that made for an awkward api for widget authors.  Most of the time you're navigating within the same app, so a widget that needs to do some navigation first has to figure out what app it's currently in, get the slug for that app, and then prepend that to the page slug.